pkg/pipelines/environments/environments.go

```diff
@@ -227,7 +227,7 @@ func ListFiles(fs afero.Fs, base string) (StringSet, error) {
 		if info.IsDir() {
 			return nil
 		}
-		filename := strings.TrimPrefix(path, base+"/")
+		filename := strings.TrimPrefix(filepath.ToSlash(path), filepath.ToSlash(base)+"/")
```

Comment:
@keithchong Thanks for the fix. We're trimming `<base>/` from the path here. Is it fine to keep the "/" hardcoded since it's direction can change with the platform?

Reply:
Yeah we need the trailing slash because the value of base that gets passed in doesn't currently have a trailing slash. (I'm not changing that behavior)

Take for example, on Windows
path is `config\cicd\base\01-namespaces\cicd-environment.yaml`

The `ToSlash` call will change this to
`config/cicd/base/01-namespaces/cicd-environment.yaml`
(on non-Windows platforms, it will be the same)

so the second argument to `TrimPrefix` must be 
`config/cicd/base/`     (with the trailing slash)

The result for `filename` will then be: `01-namespaces/cicd-environment.yaml`
and that is what we want.

Otherwise it will be `/01-namespaces/cicd-environment.yaml`

The kustomize file will have:

```
resources:
- 01-namespaces/cicd-environment.yaml
- 02-rolebindings/argocd-admin.yaml
- .....

```

Reply:
Please comment the code.  It is not straight forward to see why ToSlash() is needed.

Reply:
Please add a unit test as well.

Reply:
The unit test has been added.